woocommerce/woocommerce
# Add dimensions to shipping section

Change the FormComponent to support input name with dot notation like dimensions.width Add the dimension controls to the product form

### All Submissions:






### Changes proposed in this Pull Request:

- The form now support dot notation in input names. It uses `get` and `setWith` function from lodash.
- The @woocommerce/number now implements parseNumber function which is helpful to parse a given formatted string using the separators configured in settings into a fixed number
- An entire Dimensions Section is added to a new Product page. 
- Add validations to the size fields.
- Add number format to the fields.



Partialy closes https://github.com/woocommerce/woocommerce/issues/34329.

### How to test the changes in this Pull Request:

1. Visit the New Product page
2. A dimensions section should be shown with Width, Length, Height and Weight fields.
<img width="1073" alt="image" src="https://user-images.githubusercontent.com/13334210/192602530-d81aa7f5-0763-479f-b088-302a5233f630.png">
3. If you type a number in any of the fields, that number should be formated acording the Currency Options configured in WooCommerce > Settings > General section.
<img width="1073" alt="image" src="https://user-images.githubusercontent.com/13334210/192602717-118a7541-42f8-44c3-a910-c6302c84f3b1.png">
4. If any values is less or equals to zero an error message should be shown below the input field.
<img width="1073" alt="image" src="https://user-images.githubusercontent.com/13334210/192602792-43e3763a-9090-4e6a-93d1-eaf1c86db923.png">
5. If the product is saved successfuly with dimensions set then in the edition page the dimensions fields should show the values formatted acording the Currency Options configured in WooCommerce > Settings > General section.
<img width="1194" alt="image" src="https://user-images.githubusercontent.com/13334210/192602939-8a40f939-ad6b-4f34-acd4-93787544fa3a.png">

### Other information:




### FOR PR REVIEWER ONLY:
